src/auth/auth.js

```diff
@@ -0,0 +1,66 @@
+import express from 'express';
+import passport from 'passport';
+import * as oauth2 from 'passport-oauth2';
+
+export default function authentication(config) {
+  const app = express();
+
+  passport.use(new oauth2.Strategy({
+    authorizationURL: config.oauthAuthorizationURL,
+    tokenURL: config.oauthTokenURL,
+    clientID: config.oauthClientID,
+    clientSecret: config.oauthClientSecret,
+    callbackURL: config.serverRootURL + '/auth/login/callback'
+  },
+    (accessToken, refreshToken, profile, cb) => {
+      return cb(null, {accessToken, refreshToken, profile});
+    }
+  ));
+
+  app.use(passport.initialize());
+  app.use(passport.session());
+
+  passport.serializeUser((user, cb) => {
+    cb(null, user.accessToken);
+  });
+
+  passport.deserializeUser((accessToken, cb) => {
+    cb(null, {accessToken});
+  });
+
+  app.get('/auth/login', passport.authenticate('oauth2', {
+    scope: [
```

Comment:
You do not need to pass the scopes here. If you simplify it to:

```
app.get('/auth/login', passport.authenticate('oauth2'));
```

...then it will pull the scopes from the uaa-client's list of allowed scopes, which will mean you don't have to duplicate the scopes here as well as in the uaa-client config.